pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go

```diff
@@ -552,7 +553,18 @@ func (ds *DistSender) singleRangeFeed(
 	defer finishCatchupScan()
 
 	stuckWatcher := newStuckRangeFeedCanceler(cancelFeed, func() time.Duration {
-		return rangefeedRangeStuckThreshold.Get(&ds.st.SV)
+		// Before the introduction of kv.rangefeed.range_stuck_threshold = 1m,
+		// clusters may already have kv.closed_timestamp.side_transport_interval set
+		// to >1m. This would cause rangefeeds to continually restart. We therefore
+		// conservatively use the highest value.
+		threshold := rangefeedRangeStuckThreshold.Get(&ds.st.SV)
+		if threshold > 0 {
+			if t := time.Duration(math.Round(
+				1.2 * float64(closedts.SideTransportCloseInterval.Get(&ds.st.SV)))); t > threshold {
+				threshold = t
```

Comment:
perhaps log info? Or log.Every or some such if the settings are a bit strange?

Reply:
I would also consider adding sanity boundaries; like -- if stack threshold set to 10ms and restart set to 200 ms; perhaps having min value makes sense -- though I suppose doing the above takes active desire to make things crazy...

Reply:
`log.Info` is going to spam a ton of messages every time we set up a rangefeed. Ideally we'd want to validate this when the setting changes, but we can't do that when adding a non-zero default, and I don't think there's a way to add warnings for settings.

I think this is fine. Most people will just leave this at the default value and not care about it, and the ones who do can adjust the settings as needed.